tests/testdiscovery.nim

```diff
@@ -18,7 +18,7 @@ import
     discovery/discoverymngr,
     discovery/rendezvousinterface,
   ]
-import ./helpers
+import ./helpers, ./asyncunit, ./utils/[async, assertions, futures]
```

Comment:
Those tests need to be moved to a new PR with a different scope as they are unrelated to GossipSub.

Reply:
I don't think I follow. Why are you linking the comment to the imports?

Reply:
Sorry, I mean the tests in this file.

Reply:
I would say they are related to Gossipsub given they are specified in Gossipsub Testplan's Peer Discovery block, as linked in the description.

Do you perhaps mean to move them to a different file?

Reply:
Peer discovery isn't related to GossipSub.

Reply:
I recommend moving those tests to Rendezvous test plans and moving them to a separate PR for clarity.

Reply:
I'd argue they are related, since Peer Discovery is a dependecy of GossipSub, as clearly stated in the [official documentation](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md).
I don't consider I have any say over the test plans so, if you feel like it's inapropriate, I'd suggest raising a question in our related channel.
Having said this, I don't see the benefit of splitting this small amount of code into a different PR.

Reply:
> Whenever a new peer is connected, the gossipsub implementation checks to see if the peer implements floodsub and/or gossipsub, and if so, it sends it a hello packet that announces the topics that it is currently subscribing to.

I don't know if we are doing this part _checks to see if the peer implements floodsub and/or gossipsub_.

Reply:
Those are two different things, I believe.
What you mention is the callback that lets you run code when a peer joins or leaves. 
This is a discovery mechanism that allows you to find peers available for connection.

Reply:
Also, I believe that having removed the `xasyncTest`, this PR's scope is just peer discovery.